src/server/search/doc_index.h

```diff
@@ -30,10 +30,21 @@ struct SerializedSearchDoc {
 };
 
 struct SearchResult {
+  SearchResult() = default;
+
+  SearchResult(std::vector<SerializedSearchDoc> docs, size_t total_hits,
+               std::optional<search::AlgorithmProfile> profile)
+      : docs{std::move(docs)}, total_hits{total_hits}, profile{std::move(profile)} {
+  }
+
+  SearchResult(facade::ErrorReply error) : error{std::move(error)} {
+  }
```

Comment:
I don't like this part, it should be an io::Result<SearchResult, ErrorReply> but I'm going to expand in that area soon to optimize result serialization, so it will change and I don't see a reason to heavily invest into elegance right now